### PR TITLE
fix: exclude ghost and anonymous user when counting users

### LINKED ISSUE #3349: 后台用户显示错误
### 是什么版本出现了此问题？

2.2.1

### 使用的什么数据库？

H2

### 使用的哪种方式部署？

Docker

### 在线站点地址

_No response_

### 发生了什么？

image
image

### 相关日志输出

_No response_

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by JustinLiang522**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
后台进行用户数量统计时，排除ghost和anonymousUser

#### Which issue(s) this PR fixes:

Fixes #3349 

#### Does this PR introduce a user-facing change?

```release-note
修复仪表盘用户数显示错误问题
```

### CODE REVIEW

File: src/main/java/run/halo/app/core/extension/endpoint/StatsEndpoint.java [hunk 1]
```diff
@@ -54,7 +55,12 @@ Mono<ServerResponse> getStats(ServerRequest request) {
                 return stats;
             })
             .flatMap(stats -> client.list(User.class,
-                    user -> user.getMetadata().getDeletionTimestamp() == null,
+                    user -> {
+                        var labels = user.getMetadata().getLabels();
```
Comment (guqing) [inline src/main/java/run/halo/app/core/extension/endpoint/StatsEndpoint.java:59]: 这里建议：
```java
var labels = ExtensionUtil.nullSafeLabels(user);
return user.getMetadata().getDeletionTimestamp() == null
 && isFalse(labels.get(User.HIDDEN_USER_LABEL)) // commons lang3 BooleanUtils.isFalse
```

**Comment by guqing** [comment]
@JustinLiang522 需要解决一下冲突及改进建议

**Review by JohnNiang — APPROVED** [review]
It works very well.

image

image

/approve

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**